src/API/Site/Controllers/MerchantCenter/ShippingRateSuggestionsController.php

```diff
@@ -89,16 +97,22 @@ protected function get_suggestions_callback(): callable {
 			$country_codes = $request->get_param( 'country_codes' );
 			$rates_output  = [];
 			foreach ( $country_codes as $country_code ) {
-				$suggestions = $this->shipping_zone->get_shipping_rates_for_country( $country_code );
+				$suggestions = $this->shipping_zone->get_shipping_rates_grouped_by_country( $country_code );
 
 				// Prepare the output.
 				$suggestions = array_map(
-					function ( $rate ) use ( $request ) {
-						// Remove the shipping class rates from the response because they are not needed.
-						if ( isset( $rate['options']['shipping_class_rates'] ) ) {
-							unset( $rate['options']['shipping_class_rates'] );
+					function ( $location_rate ) use ( $request ) {
+						$item = $location_rate->jsonSerialize();
+
+						// Add the store currency to each rate.
+						$item['currency'] = $this->wc->get_woocommerce_currency();
+
+						// Set the rate to 0 if it is not set.
+						if ( ! isset( $item['rate'] ) ) {
+							$item['rate'] = 0;
```

Comment:
I've added two commits to fix these. This rate should be set to a float. And the comparison is now changed to "> 0" so that would avoid any type checking issues. In general, the rate is always returned as a float but since we're passing them around as arrays it's not possible to type check them everywhere. I aim to change this in the next PR by using the value objects introduced in this PR.